Matey/Controllers/AddNewTransactionViewController.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  AddNewTransactionViewController.swift
+//  Matey
+//
+//  Created by Firat Polat on 30.08.2022.
+//
+
+import Foundation
+import UIKit
+
+protocol AddNewTransactionViewControllerInterface: AnyObject {
+    func uiInit()
+}
+
+final class AddNewTransactionViewController: UIViewController {
+
+    @IBOutlet weak private var textFieldRegistrantsName: UITextField!
+    @IBOutlet weak private var textFieldRegistrantsUsername: UITextField!
+    @IBOutlet weak private var textFieldFriendUsername: UITextField!
+    @IBOutlet weak private var textFieldLendAmount: UITextField!
+    @IBOutlet weak private var textFieldBorrowAmount: UITextField!
+    @IBOutlet weak private var buttonSaveNewTransaction: UIButton!
+
+    var viewModel: AddNewTransactionViewModelInterface?
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        viewModel = AddNewTransactionViewModel(view: self)
+        viewModel?.notifyViewDidload()
+    }
+
+    // MARK: - When user pressed the save button
+
+    @IBAction func saveTransactionButtonAction(_ sender: UIButton) {
+        if validationTextFields() {
+            viewModel?.saveTransaction(name: textFieldRegistrantsName.text ?? "", friend: textFieldFriendUsername.text ?? "", lend: textFieldLendAmount.text ?? "0.0", borrow: textFieldBorrowAmount.text ?? "0.0", username: textFieldRegistrantsUsername.text ?? "")
+            showAlert(title: "Saved", message: "Transaction saved successfully!")
+            setupTextFieldEmpty()
+        }
+    }
+
+    // MARK: - Validation func for textfields
+
+    private func validationTextFields() -> Bool {
+        if textFieldRegistrantsName.text == "" {
```

Comment:
```suggestion
        if textFieldRegistrantsName.text.isEmpty {
```
is clean than that, u can apply it to others.